lib/modules/datasource/datasource.ts

```diff
@@ -19,7 +19,7 @@ export abstract class Datasource implements DatasourceApi {
 
   defaultConfig: Record<string, unknown> | undefined;
 
-  defaultRegistryUrls: string[] | undefined;
+  defaultRegistryUrls: string[] | (() => string[]) | undefined;
```

Comment:
```suggestion
  defaultRegistryUrls?: string[] | (() => string[]);
```

Works too and looks cleaner to me

Reply:
Sorry! I made changes locally need to push them